corehq/apps/reports/analytics/dbaccessors.py

```diff
@@ -10,6 +10,7 @@ def get_wrapped_ledger_values(domain, case_ids, section_id, entry_ids=None):
         case_ids_list = list(case_ids)
     else:
         case_ids_list = [case_ids]
+    entry_ids = entry_ids or None
```

Comment:
Are you sure about this behavior?  I could imagine a scenario where `entry_ids=[]` returning results for all products would be confusing (though it'd make sense at the report level).  I'm not familiar with how this is used, but it feels a little nebulous.

Reply:
This change is because the couch and sql backends have a slightly different check:

SQL checks from NULL:

https://github.com/dimagi/commcare-hq/blob/00e0292771e7d7ed1d426e580759304017bc07ee/corehq/form_processor/backends/couch/dbaccessors.py#L360-L362

But couch just checks for 'falsy':

https://github.com/dimagi/commcare-hq/blob/2f54a1b6f60cc128f243f0e0449ee05af0644fbd/corehq/sql_accessors/sql_templates/get_ledger_values_for_cases_2.sql#L21-L23

Perhaps the better place to change this would be in the sql function:
```SQL
IF p_entry_ids IS NOT NULL AND array_length(p_entry_ids, 1) > 0 THEN
        select_expr := select_expr || entry_filter;
END IF;
```

Reply:
Yeah, I like that - definitely seems like the two backends should have the same behavior in that regard, if possible to safely realign them at this stage.  If changing the SQL function is hard, you could also coerce `[]` to `None` in the python code that calls it.

Reply:
Updated sql functions